pkg/lib/location.ts

```diff
@@ -37,7 +37,7 @@ export function calculate_application() {
     return path.split("/")[1];
 }
 
-export function calculate_url(suffix) {
+export function calculate_url(suffix?: string) {
```

Comment:
This is the only change for location.ts typescriptification. It also exports calculate_application(), but that's not annotated (with `=> string`). Do you rely on implicit return type calculation of tsc? TBH I don't like that much -- I find it better to pin down the API and say "we always return a string", and let tsc tell me which exit paths I messed up, instead of silently assuming the function returns something like `string | UInt8Array | null`.

In general: If we do this exercise, IMHO we should annotate all functions -- even with `unknown` for difficult cases, that's much easier to grep than "absent"

Reply:
Ya.  I think you changed my mind about this yesterday, but I was already in the middle of this PR.  I'll go back and add them :)

Reply:
Done.